src/App.js

```diff
@@ -1,19 +1,178 @@
 import React from 'react';
 import './App.css';
+import { ToDoList } from './components/ToDoList';
 
 import users from './api/users';
+import todos from './api/todos';
 
-function App() {
-  return (
-    <div className="App">
-      <h1>Add todo form</h1>
-
-      <p>
-        <span>Users: </span>
-        {users.length}
-      </p>
-    </div>
-  );
+const prepearedTodos = users.map(user => ({
+  ...user,
+  todo: todos.filter(todo => user.id === todo.userId),
+}));
+
+class App extends React.Component {
+  state = {
+    listOfTodos: [...prepearedTodos],
+    user: '',
+    todo: '',
+    inValidSelect: false,
+    inValidLength: false,
+    isTooLong: false,
+    containsSymbols: false,
+    containsWords: true,
+  }
+
+  changeInSelect(e) {
+    this.setState({
+      user: e.target.value,
+      inValidSelect: false,
+    });
+  }
+
+  formSubmit(e, user, todo, containsSymbols, isTooLong, containsWords) {
+    e.preventDefault();
+    if (user.length > 0 && todo.length > 0
+      && !containsSymbols && !isTooLong && !containsWords) {
+      this.setState({
+        user: '',
+        todo: '',
+      });
+    }
+  }
+
+  changeInput(e) {
+    if (e.target.value.length > 10) {
+      this.setState({
+        isTooLong: true,
+      });
+    } else {
+      this.setState({
+        isTooLong: false,
+      });
+    }
+
+    if (/[^\w\s]/.test(e.target.value)) {
+      this.setState({
+        containsSymbols: true,
+      });
+    } else {
+      this.setState({
+        containsSymbols: false,
+      });
+    }
+
+    if (e.target.value.replace(/ /g, '').replace(/[0-9]/g, '').length === 0) {
+      this.setState({
+        containsWords: false,
+      });
+    } else {
+      this.setState({
+        containsWords: true,
+      });
+    }
+
+    this.setState({
+      todo: e.target.value,
+      inValidLength: false,
+    });
+  }
+
+  clickOnButton(todo, user, listOfTodos, isTooLong,
+    containsSymbols, containsWords) {
+    if (user.length > 0 && todo.length > 0 && !isTooLong
+      && !containsSymbols && containsWords) {
+      const targetUser = listOfTodos.find(person => (
+        person.name === user
+      ));
+
+      if (targetUser) {
+        targetUser.todo.push({
+          userId: targetUser.id,
+          id: Math.floor(Math.random() * 100),
+          title: todo,
+          completed: false,
+        });
+      }
+
+      this.forceUpdate();
+    }
```

Comment:
you need to update a list of todos, not users
also we update state ONLY with setState, then React will automatically rerender the UI (no need to forceUpdate)